Source/JNA/waffle-tests/src/test/java/waffle/jaas/WindowsLoginModuleTests.java

```diff
@@ -291,17 +275,10 @@ void testRoleUnique() throws LoginException {
         Assertions.assertTrue(this.loginModule.login());
         Assertions.assertTrue(this.loginModule.commit());
 
-        Assertions.assertEquals(2, subject.getPrincipals().size());
-        Assertions.assertTrue(subject.getPrincipals().contains(new GroupPrincipal("Roles")));
-        for (final Principal principal : subject.getPrincipals()) {
-            if (principal instanceof GroupPrincipal) {
-                int size = Collections.list(((GroupPrincipal) principal).members()).size();
-                Assertions.assertEquals(3, size);
-                Assertions.assertTrue(((GroupPrincipal) principal).isMember(new RolePrincipal("Everyone")));
-                Assertions.assertTrue(((GroupPrincipal) principal).isMember(new RolePrincipal("Users")));
-                Assertions.assertTrue(((GroupPrincipal) principal).isMember(new RolePrincipal("Group 1")));
-            }
-        }
+        Assertions.assertTrue(4, subject.getPrincipals().size() >= 4);
```

Comment:
@eekodeerder Can you fix line 278 here?  Needs the 4 on left side removed.  I think everything will work then.

Reply:
Sorry.  I booked it in (under draft PR).  Intntion is/was to come back and fix errors as soon as I get time.  I also fixed the test that was failing, but for some reason Tomcat 10 with JDK 8 and windows-2016 is failing to build.